mozci/data/sources/treeherder/__init__.py

```diff
@@ -103,6 +107,7 @@ def normalize(cls, jobs):
                 "success": "passed",
                 "testfailed": "failed",
                 "bustage": "failed",
```

Comment:
Do we also need to keep `bustage` in this map or is `busted` enough?

Reply:
`bustage` never comes into Treeherder.  Not sure if that's needed for something else though.

> Nit: It'd be nice to split the PR into four commits:
> 
>     * one to add the "state" field;
> 
>     * one to avoid setting "result" when the task is unscheduled or still running;
> 
>     * one to add the implementation of the `push_test_groups` contract;
> 
>     * one to improve the logging in the **post_init** function of the TestTask class.

Sure, I can split it to those commits.  👍

Reply:
Done

Reply:
I've checked in ActiveData too, the `bustage` string never occurs. Probably a typo by @ahal, so we can remove it altogether (I'll remove it from the ActiveData source too).